library/platform_util.c

```diff
@@ -26,18 +26,44 @@
 #define _POSIX_C_SOURCE 200112L
 #endif
 
+#if !defined(_GNU_SOURCE)
+/* Clang requires this to get support for explicit_bzero */
+#define _GNU_SOURCE
+#endif
+
 #include "common.h"
 
 #include "mbedtls/platform_util.h"
 #include "mbedtls/platform.h"
 #include "mbedtls/threading.h"
 
 #include <stddef.h>
+
+#ifndef __STDC_WANT_LIB_EXT1__
+#define __STDC_WANT_LIB_EXT1__ 1 /* Ask for the C11 gmtime_s() and memset_s() if available */
+#endif
 #include <string.h>
 
+#if defined(_WIN32)
+#include <windows.h>
+#endif
+
+// Detect platforms known to support explicit_bzero()
+#if defined(__GLIBC__) && (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 25) \
+    && !defined(__ARM_EABI__)
+#define MBEDTLS_PLATFORM_HAS_EXPLICIT_BZERO 1
+#elif defined(__FreeBSD__) && (__FreeBSD_version >= 1100037)
```

Comment:
I'd be very happy to see OpenBSD support here, but sInce `explicit_bzero()` went into OpenBSD 9 years ago, and OpenBSD typically only supports release for a year, we can assume that it is present, there's really no point to the `201405` check - just assume it's present. Anyone building on OpenBSD without `explicit_bzero()` is on an ancient system that went out of support aeons ago

Reply:
We don't have OpenBSD on our CI, but we have included feature checks for other OSses in the past. I can't easily test this one at the moment, and I'm not sure this is actually right (as opposed to needing an extra `#define _BSD_SOURCE` or something. Since you can't easily test it either, please don't include it.

(This is really more me expressing disappointment at the quality of OpenBSD documentation.)

Reply:
Re `__BSD_VISIBLE` I suspect that gets defined automatically during builds... (I don't have a viable system here right now :( )

Reply:
[It does](https://cvsweb.openbsd.org/cgi-bin/cvsweb/src/sys/sys/cdefs.h?rev=1.43&content-type=text/x-cvsweb-markup), based on various other symbols like `_POSIX_xxx` and such. This doesn't result in a clear way to determine whether a prototype for a given function is available.

Reply:
```
openbsd-vm $ uname -a
OpenBSD openbsd-vm.local 7.2 RAMDISK#1719 arm64

openbsd-vm $ cat foo.c
#include <string.h>

int main(int argc, char *argv[])
{
    explicit_bzero(argv[0], 1);
    return 0;
}

openbsd-vm $ cc -o foo -Wall -Wextra -Wno-unused-parameter -Werror foo.c

openbsd-vm $ echo $?
0

openbsd $ echo | cc -dM -E - | fgrep OpenBSD
#define __OpenBSD__ 1
#define __VERSION__ "OpenBSD Clang 13.0.0"
```

Reply:
i.e. detect with `#if defined(__OpenBSD__)`, `#include <string.h>` and then just call `explicit_bzero()`

Reply:
Done in https://github.com/Mbed-TLS/mbedtls/pull/7220.